specs/select.spec.md

```diff
@@ -0,0 +1,156 @@
+# Select + Select-Option
+
+## Overview
+
+The `select` is a component that provides a list of options for the user to select from.
+
+### Background
+
+There is a standard select html control, `<select>` and `<option>`, but it doesn't support the full level of detail, styling, etc that most design systems need or require. For context please see [MDN Docs](https://developer.mozilla.org/en-US/docs/web/html/element/select).
+
+### Use Cases
+
+- *A customer using the component on a web page.*
+On a web page a customer is shopping for a new shirt in size medium. The customer opens the size `select` and sees a list of `<option>`s and selects size medium.
+
+### Features
+- form association
+- focus delegation
+
+### Features
+- Select one or multiple `<fast-option/>`(s) using the `multiple` attribute
+- When Dropdown does not have enough screen real estate to open below `<fast-select>` it will open above.
+
+### Prior Art/Examples
+- [FAST Select (React)](https://www.npmjs.com/package/@microsoft/fast-components-react-msft)
+- [Ant Design](https://ant.design/components/select/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/select/code/)
+- [Atlassian UI](https://atlaskit.atlassian.com/packages/core/select)
+- [Lightning Design System](https://www.lightningdesignsystem.com/components/select/)
+- [W3 Example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html)
+
+---
+
+### API
+
+*Component Name*:
+- `fast-select`
+
+*Attributes:*
+- `autocomplete` - Allows the developer to provide a hint on how to search the content within the `<fast-option>`(s)
```

Comment:
Would be great to see a more through explanation of how this is intended to work.

Reply:
I was following the Open UI spec, I believe in the past we filtered the potential results, but thinking about this I'm a bit confused on why select would have this. This seems like more of feature of a dropdown on an autosuggest of textbox. @gregwhitworth can you clarify this feature?

Reply:
We can also hold on adding autocomplete in the initial implementation.

Reply:
Actually this may be a good thing to move to "next steps" - investigate autocomplete, etc...

Reply:
@gregwhitworth and I were having a discussion offline and wanted to include those discussions here. 

We will move this to next steps but wondering if select should have a text input or not, This seems more like a combobox or search box to me. I could not find any documentation online that "searching" options should be part of a select. 

My issue with using the terminology of "search" is that when you search the text you input is not always in the suggested dropdown, which makes it not a select. Because you are not selecting some predetermined value